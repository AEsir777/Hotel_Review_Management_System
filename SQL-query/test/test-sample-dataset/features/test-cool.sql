/****** cool ******/
CALL CoolByRid('R9', 'U8');
CALL CoolByRid('U1', 'R9');
CALL CoolByRid('R1', 'U8');
CALL CoolByRid('R1', 'R9');
CALL CoolByRid('R1', 'U1');
CALL CoolByRid('R3', 'R1');

/* test if a review is already cooled  */
CALL IsCool('R9', 'U8');
CALL IsCool('R1', 'U1');
CALL IsCool('R032', 'U432');
CALL IsCool(null, 'U3');

/* test if cools added in user profile */
CALL GetUserByUid('U1');
CALL GetUserByUid('U8');

/* cancel cool */
CALL cancelCool('R9', 'U8')
CALL cancelCool('R1', 'U1')
CALL cancelCool('R3', 'U1')

/* test if cools in user profile update after cool got cancelled*/
CALL GetUserByUid('U1');
CALL GetUserByUid('U8');