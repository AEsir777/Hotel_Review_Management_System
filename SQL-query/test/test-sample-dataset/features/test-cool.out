/****** cool ******/
// CALL CoolByRid('R9', 'U8');
'R9', 'U8'
// Follow up: SELECT * FROM CoolHistory WHERE uid = 'U8' AND rid = 'r1';
// expected output: 'R9', 'U8'

// CALL CoolByRid('R9', 'U1');
'R9', 'U1'
// Follow up: SELECT * FROM CoolHistory WHERE uid = 'U1' AND rid = 'R9';
// expected output: 'R9', 'U1'



/* test if a review is already cooled  */
// CALL IsCool('R9', 'U8');
'R9', 'U8'
// expected output: 'R9'	'U8'

// CALL IsCool('R1', 'U1');
'R1', 'U1'
// expected output: 'R1', 'U1'

// CALL IsCool('R032', 'U432');
null
// expected output: not found

// CALL IsCool(null, 'U3');
null
// expected output: not found